src/main/java/com/exactpro/th2/healer/grpc/HealerServiceImpl.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright 2020-2021 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exactpro.th2.healer.grpc;
+
+import com.exactpro.cradle.CradleStorage;
+import com.exactpro.cradle.testevents.StoredTestEventId;
+import com.exactpro.cradle.testevents.StoredTestEventWrapper;
+import com.exactpro.th2.healer.cache.EventsCache;
+import com.exactpro.th2.healer.cfg.Check2Configuration;
+import com.exactpro.th2.common.grpc.EventID;
+import com.exactpro.th2.common.grpc.EventStatus;
+import com.exactpro.th2.crawler.dataservice.grpc.CrawlerId;
+import com.exactpro.th2.crawler.dataservice.grpc.CrawlerInfo;
+import com.exactpro.th2.crawler.dataservice.grpc.DataServiceGrpc;
+import com.exactpro.th2.crawler.dataservice.grpc.DataServiceInfo;
+import com.exactpro.th2.crawler.dataservice.grpc.EventDataRequest;
+import com.exactpro.th2.crawler.dataservice.grpc.EventResponse;
+import com.exactpro.th2.crawler.dataservice.grpc.Status;
+import com.exactpro.th2.dataprovider.grpc.EventData;
+import io.grpc.stub.StreamObserver;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static com.exactpro.th2.common.message.MessageUtils.toJson;
+
+
+public class HealerServiceImpl extends DataServiceGrpc.DataServiceImplBase {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(HealerServiceImpl.class);
+
+    private final Check2Configuration configuration;
+    private final CradleStorage storage;
+    private final Map<String, InnerEvent> cache;
+    private final ConcurrentHashMap.KeySetView<CrawlerId, Boolean> knownCrawlers;
+
+    public HealerServiceImpl(Check2Configuration configuration, CradleStorage storage) {
+        this.configuration = configuration;
+        this.storage = Objects.requireNonNull(storage, "Cradle storage cannot be null");
+        this.cache = new EventsCache<>(configuration.getCacheSize());
+        this.knownCrawlers = ConcurrentHashMap.newKeySet();
+    }
+
+    @Override
+    public void crawlerConnect(CrawlerInfo request, StreamObserver<DataServiceInfo> responseObserver) {
+        try {
+            LOGGER.info("crawlerConnect request: {}", request);
```

Comment:
Please, use either `TextFormat.shortDebugString` or `MessageUtils.toJson` method for writing proto object into the log. Otherwise, they will be formatted for humans. It is not good for logging.

Also, if you will use those methods you will need to wrap the logger call in the condition

```java
if (LOGGER.isLevelEnabled()) {
  LOGGER.level("Message {}", MessageUtils.toJson(proto, true))
}
```